File: php/day5_part1_2017.php

<?php

$data = file_get_contents("input.txt");
$lines = explode("\n", trim($data));
$offsets = [];
foreach ($lines as $line) {
    $offset = (int)$line;
    $offsets[] = $offset;
}

$index = 0;
$steps = 0;

while ($index >= 0 && $index < count($offsets)) {
    $jump = $offsets[$index];
    $offsets[$index]++;

    $index += $jump;
    $steps++;
}

echo $steps . PHP_EOL;
?>

File: csharp/day5_part1_2017.cs

using System;
using System.IO;

class Program
{
    static void Main()
    {
        string[] lines = File.ReadAllLines("input.txt");
        int[] offsets = Array.ConvertAll(lines, int.Parse);

        int index = 0;
        int steps = 0;

        while (index >= 0 && index < offsets.Length)
        {
            int jump = offsets[index];
            offsets[index]++;
            index += jump;
            steps++;
        }

        Console.WriteLine(steps);
    }
}
